packages/react-components/src/components/chart/converters/convertSeriesAndYAxis.ts

```diff
@@ -204,12 +204,15 @@ export const useSeriesAndYAxis = (
   const defaultYAxis: YAXisComponentOption[] = useMemo(() => [convertChartYAxis(axis)], [axis]);
   const convertedThresholds = convertThresholds(thresholds);
   const getStyles = getChartStyleSettingsFromMap(styleSettings);
+  const datastreamDeps = JSON.stringify(datastreams.map(({ id, refId }) => `${id}-${refId}`));
 
   const { series, yAxis, yMaxs, yMins } = useMemo(() => {
     return datastreams
       .map((datastream) => convertSeriesAndYAxis(getStyles(datastream))(datastream))
       .reduce(reduceSeriesAndYAxis, { series: [], yAxis: defaultYAxis, yMins: [], yMaxs: [] });
-  }, [datastreams, styleSettings, defaultYAxis]);
+    // disabling because datastream are stringified as datastreamDeps
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
I dont know what this exactly means, but looks like its worth adding an exception / turning off this rule in the plugin instead of having this eslint-disable comment all over the place?

Reply:
in this case the eslint rule doesnt realize that `datastreamDeps` represents the `dataStreams` object
we dont want to turn this rule off everywhere though because it does apply in other situations